torch/csrc/jit/tensorexpr/llvm_jit.cpp

```diff
@@ -38,32 +38,61 @@ static llvm::JITTargetAddress toAddress(T* Ptr) {
   return static_cast<llvm::JITTargetAddress>(reinterpret_cast<uintptr_t>(Ptr));
 }
 
-static llvm::orc::JITTargetMachineBuilder makeTargetMachineBuilder() {
-#if 0
-  // FIXME: Switch to using detectHost() rather than setting up the JTMB manually
-  // once LLVM 10 is available.
-  return assertSuccess(llvm::orc::JITTargetMachineBuilder::detectHost());
-#else
+// Get subtarget features for the host.
+static llvm::SubtargetFeatures getHostSubtargetFeatures() {
+  llvm::SubtargetFeatures subtargetFeatures;
+  llvm::StringMap<bool> featureMap;
+  llvm::sys::getHostCPUFeatures(featureMap);
+  for (auto& feature : featureMap) {
+    subtargetFeatures.AddFeature(feature.first(), feature.second);
+  }
+  return subtargetFeatures;
+}
+
+// Create a JTMB using the host's triple.  CPU and attrs default to the host
+// unless they are supplied.
+static llvm::orc::JITTargetMachineBuilder makeJTMBFromHost(
+    c10::optional<std::string> cpu,
+    c10::optional<std::string> attrs) {
```

Comment:
curious: when do we want to override the host cpu/attrs?

Reply:
Good question.  I'm not sure whether this would be useful for mobile builds, since we don't (I think?) customize them at the level of cpu.  Server side it's useful to be able to specify  cpu, since you could have something like `-mtriple=x86_64-unknown-linux-gnu -mcpu=skylake-avx512` which would enable using avx512 instructions.